syslog-ng-rs/syslog-ng-common/src/proxies/parser/mod.rs

```diff
@@ -54,58 +56,123 @@ pub mod _parser_plugin {
 
     #[no_mangle]
     pub extern fn native_parser_proxy_init(this: &mut ParserProxy<$name>) -> c_int {
-        let result = this.init();
-        bool_to_int(result)
+        let mut wrapper = AssertUnwindSafe(this);
+
+        match catch_unwind(move || wrapper.init()) {
+            Ok(init_result) => bool_to_int(init_result),
```

Comment:
Could you move all such `bool_to_int()` to inside the `catch_unwind()`? This would enable introducing a pattern for the whole boilerplate of `match catch_unwind(...) { Ok(X)=>X, Err(MSG) => ...}.